checkr/src/pg.rs

```diff
@@ -199,12 +201,23 @@ impl Command {
     }
 }
 
-fn done(guards: &[Guard]) -> BExpr {
-    guards
-        .iter()
-        .map(|g| BExpr::Not(Box::new(g.0.clone())))
-        .reduce(|a, b| BExpr::logic(a, LogicOp::And, b))
-        .unwrap_or(BExpr::Bool(true))
+fn done(det: Determinism, guards: &[Guard]) -> BExpr {
```

Comment:
Regarding the done function, I suspect there is a minor difference between how the two are calculated. The non-determinstic version is on page 19 of Formal Methods and the deterministic version of it is on page 25 of Formal Methods.

It would appear, that in the non-deterministic version, they extract each b from the guards, `not` them, and then `and` them, yielding a `done` pattern as follows: `-b1^-b2^...^-bn`, whereas in the deterministic version, they can be seen to do `-d` instead (where `d` is the output of the deterministic edges function), which would as a result yield the pattern: `-(b1 v b2 v ... v bn)`. In essence, they express exactly the same when evaluating their truth tables, but if one was to go by these definitions, they would seemingly still be expressed differently. Though I can easily see that having a determinism argument in `done` seems a bit odd, as none of the definitions use determinism explicitly in their definitions, though I'm a bit unsure, whether it would be best to go strictly by the Page 19 and 25 definitions, or if only one of the `done` methods should be chosen?

Reply:
Good points! It essentially applies De Morgan's laws to `done`, which is a bit arbitrary 🤔